src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -137,6 +156,62 @@ def signal_handler(sig, frame):
     raise KeyboardInterrupt
 
 
+def log_stack_recursively(task: asyncio.Task, elapse_time: float):
+    """Recursively log the frame of a task or coroutine.
+    Traditional stacktrace would stop at the first awaited nested inside the coroutine.
+
+    :param task: Task to log
+    :type task_or_coroutine: asyncio.Task
+    :param elapse_time: Seconds elapsed since the task started
+    :type elapse_time: float
+    """
+    messages = [f"Task {task.get_name()} has been running for {elapse_time:.0f} seconds, stacktrace:"]
+    task_or_coroutine = task
+    while True:
+        if isinstance(task_or_coroutine, asyncio.Task):
+            # For a task, get the coroutine it's running
+            coroutine: asyncio.coroutine = task_or_coroutine.get_coro()
+        elif asyncio.iscoroutine(task_or_coroutine):
+            coroutine = task_or_coroutine
+        else:
+            flow_logger.info("\n".join(messages))
```

Comment:
I remember thread version is printing warning instead of info?

Reply:
Good catch. updated.